### DISCUSSION TITLE
Is it possible to add step and reverse operations to the range loop

### DISCUSSION THREAD

**Discussion by tzSharing** [vlang/v#13126] [Ideas]
**Like this：**

- **Step op**
```
for i in 0 .. 8 step 2{
    print(i)       // Prints '0246'
}
```

- **Reverse op**
```
for i in 8 .. 0{
    print(i)
}

or

for i in 0 .. 8 rev{
    print(i)
}
```

**Comment by ghost** [comment]
This has been requested many times here and there, but Alex has rejected them all, so unfortunately it will never be implemented.

**Reply by tzSharing** [reply]
> This has been requested many times here and there, but Alex has rejected them all, so unfortunately it will never be implemented.

Got it, but it doesn't matter, the most important thing at this stage is to release version 0.3

**Comment by JalonSolov** [comment]
https://github.com/Delta456/range

**Comment by JalonSolov** [comment]
The main reason these won't be added to V itself is that they're really just syntactic sugar over the C-style `for` loop V already supports.  Not only does this go against the V "one way" philosophy, it is "costly" to add new keywords to the language, especially for something that's already supported in a slightly different way.

**Comment by Wajinn** [comment]
Maybe it would be good to show some examples for others new to V that might come by this.

```
fn main() {
	loop_play()
}

fn loop_play() {
	mut colors := ['green','blue','red','purple','orange','yellow']
	mut new_colors := []string{}
	mut index := 0
	
	println('show color array: $colors')
	
	for index = 0; index < colors.len ; index++  {
		println('normal order of colors: ' + colors[index].str())
	}
	
	for index = colors.len - 1; index >= 0; index--  {
		println('reverse order of colors: ' + colors[index].str())
	}
	
	for index = colors.len - 1; index >= 0; index -= 2  {
		println('reverse order and skipping of colors: ' + colors[index].str())
	}
	
	println('reverse colors: ' + colors.reverse().str())  // reverse elements of string array
	
	for value in colors[2..5]  {
		println('subrange of colors: $value')
	}
	
	for key, value in colors[1..6] {
		if key % 2 == 0 {
			new_colors << value 
		}
	}
	println('every other color: $new_colors')
	println ('reverse of new_color array: ' + new_colors.reverse().str())
}
```
